src/components/CardsContainer.tsx

```diff
@@ -1,28 +1,83 @@
 'use client'
 
 import UserCard from './UserCard'
+import { useState, useEffect } from 'react'
+
 import { namesArray } from '@/utils/sampleData'
 import { sealifeImages } from '@/utils/imageLists'
+// import ChatQuestionOutline from './icons/JellyFishOutlineIcon'
+import { useAllUsersPointsListener } from '@/services/socket'
+import { useSocketListener } from '@/services/socket'
+import type { ListenerRes } from '@/services/socket'
 
 export default function CardsContainer() {
-	const sampleNames = namesArray.slice(10, 30)
+	const [usersPointsData, setUsersPointsData] = useState<ListenerRes[]>([])
+	const [showStoryPoints, setShowStoryPoints] = useState(false)
+	// const sampleNames = namesArray.slice(0, 9)
+	const sampleNames: string[] = []
 	const storyPointsArray = [0, 1, 2, 3, 5, 8, 13]
 
+	const allUsersPoints = useAllUsersPointsListener('all-users-story-points')
+
+	// the line below creates an infinite loop
+	// allUsersPoints && setUsersPointsData(allUsersPoints?.message)
+
+	const showStoryPointsListener = useSocketListener(
+		'show-disable-reset-points',
+	)
+
+	useEffect(() => {
```

Comment:
todo: This 2 `useEffect` can be replaced by the socket callback pattern we spoke in our call :)

Reply:
DONE

Reply:
todo: Have you realized in this specific listener, you don't even need the callback nor the state? This is all you need:

```js
const allUsersStoryPoints = useAllUsersPointsListener('all-users-story-points');
const usersPointsData = allUsersStoryPoints?.message || []
```

To compare, this is your original code:

```js
const [usersPointsData, setUsersPointsData] = useState<ListenerRes[]>([])
const allUsersPoints = useAllUsersPointsListener('all-users-story-points')

useEffect(() => {
	if (!allUsersPoints) return
	const listenerMessages = allUsersPoints.message
	console.log('%c>>> listenerMessages:', 'color: yellow', listenerMessages)
	setUsersPointsData(listenerMessages)
}, [allUsersPoints])
```


In other words: Your original implementation had 2 useEffects and 4 re-renders. This new implementation has 1 useEffect, 1-rerender and waaaay less code! 🔥 Do you see it now? :D

Reply:
I am really confused. When I tried this before, I got an infinite loop. I'm not sure what I did, exactly, I think maybe I tried to setState with `usersPointsData` and that created the infinite loop. Either way, I must not have done exactly what you have above, because that does work. Weird. That's why I had useEffect in the first place, the infinite loop. 

DONE

Reply:
If you got an infinite loop is because you were using `useState()` instead of direct assign.

Tip: When you get stuck in a version, is a good idea to commit it, and then commit again the fix. Then you can go back in history easily to compare old code 💡

Reply:
That's exactly what I was doing, useState. I get it now.